#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.9989885364374299 -0.033513946307738345 -0.02997865022785308 1.46055608237908
  position 0.03983896694096112 1.7774770500056656 0.13990600223376637
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2.1445 1.4855
  floorTileSize 0.33 0.33
}
DEF z5209457 E-puck {
  hidden position_0_0 -3.1082146394565126e-08
  hidden position_0_1 3.108176974075628e-08
  hidden rotation_1 1 0 0 2.9802322387695313e-08
  hidden rotation_2 -1 0 0 2.9802322387695313e-08
  translation 0.48986800003339565 -6.395900952206428e-05 0.3284999532751388
  rotation 5.116314072480576e-06 0.9999999999866085 7.787274615496884e-07 1.5707953318759698
  name "e-puck(1)"
  controller "z5209457_main_controller"
  camera_width 320
  camera_height 240
  turretSlot [
    Compass {
    }
    DistanceSensor {
      translation -0.035 -0.01 -0.01
      rotation 0 1 0 3.14159
      name "left sensor"
    }
    DistanceSensor {
      translation 0.035 -0.01 -0.01
      name "right sensor"
    }
    DistanceSensor {
      translation 0 -0.01 -0.045
      rotation 0 1 0 1.5708
      name "forward sensor"
    }
  ]
}
DEF MAZE Group {
  children [
    DEF cornerStone Solid {
      translation -0.7424999999999817 0.03249942620380372 -0.41249999999998116
      rotation -0.1980620183115681 0.8199130826680685 0.5371349679290051 -5.862637813342934e-16
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(4)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
      linearVelocity 2.1778743786074847e-17 -6.937815564207392e-10 1.520037192645703e-16
      angularVelocity 4.797852310393934e-15 -2.0382533184877035e-16 -1.9991051293307792e-16
    }
    DEF cornerStone Solid {
      translation 0.7425000000000184 0.03249942620380372 -0.41249999999998116
      rotation -0.1980620183115681 0.8199130826680685 0.5371349679290051 -5.862637813342934e-16
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(5)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
      linearVelocity 2.1778743786074847e-17 -6.937815564207392e-10 1.520037192645703e-16
      angularVelocity 4.797852310393934e-15 -2.0382533184877035e-16 -1.9991051293307792e-16
    }
    DEF cornerStone Solid {
      translation 0.7425000000000184 0.03249942620380372 0.4125000000000188
      rotation -0.1980620183115681 0.8199130826680685 0.5371349679290051 -5.862637813342934e-16
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(6)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
      linearVelocity 2.1778743786074847e-17 -6.937815564207392e-10 1.520037192645703e-16
      angularVelocity 4.797852310393934e-15 -2.0382533184877035e-16 -1.9991051293307792e-16
    }
    DEF cornerStone Solid {
      translation -0.7424999999999817 0.03249942620380372 0.4125000000000188
      rotation -0.1980620183115681 0.8199130826680685 0.5371349679290051 -5.862637813342934e-16
      children [
        DEF cornerStone Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 0.5
          }
          geometry Box {
            size 0.015 0.065 0.015
          }
        }
      ]
      name "solid(7)"
      boundingObject USE cornerStone
      physics Physics {
        density 100
      }
      linearVelocity 2.1778743786074847e-17 -6.937815564207392e-10 1.520037192645703e-16
      angularVelocity 4.797852310393934e-15 -2.0382533184877035e-16 -1.9991051293307792e-16
    }
    DEF hWall Solid {
      translation 6.200127079190163e-13 0.03244415049641847 -0.4124999997715785
      rotation 7.44240869873766e-08 0.9999999999999972 2.9863439363876146e-10 1.1891177945660066e-08
      children [
        DEF hWall Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 0.3
          }
          geometry Box {
            size 1.46 0.065 0.015
          }
        }
      ]
      name "solid(1)"
      boundingObject USE hWall
      physics Physics {
        density 100
      }
      linearVelocity 2.268664952507478e-18 -6.749548436997926e-08 5.819021623856189e-17
      angularVelocity 2.103168535375732e-15 -1.0263464301851264e-17 -7.011137363387554e-17
    }
    DEF hWall Solid {
      translation 6.200127079190163e-13 0.03244415049641847 0.4125000002284215
      rotation 7.44240869873766e-08 0.9999999999999972 2.9863439363876146e-10 1.1891177945660066e-08
      children [
        DEF hWall Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 0.3
          }
          geometry Box {
            size 1.46 0.065 0.015
          }
        }
      ]
      boundingObject USE hWall
      physics Physics {
        density 100
      }
      linearVelocity 2.268664952507478e-18 -6.749548436997926e-08 5.819021623856189e-17
      angularVelocity 2.103168535375732e-15 -1.0263464301851264e-17 -7.011137363387554e-17
    }
    DEF vWall Solid {
      translation -0.7424999999263405 0.03246939753407326 4.075123312693221e-13
      rotation -3.0495609000141227e-09 0.9999999999999968 -7.894460568138583e-08 -6.782941680205178e-09
      children [
        DEF vWall Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.3
          }
          geometry Box {
            size 0.015 0.065 0.8
          }
        }
      ]
      name "solid(2)"
      boundingObject USE vWall
      physics Physics {
        density 100
      }
      linearVelocity -7.52090552224987e-17 -3.69919929673794e-08 4.268010667150119e-18
      angularVelocity 2.6513024623505304e-16 -4.833467165030712e-18 2.159033281127216e-15
    }
    DEF vWall Solid {
      translation 0.7425000000736596 0.03246939753407326 4.075123312693221e-13
      rotation -3.0495609000141227e-09 0.9999999999999968 -7.894460568138583e-08 -6.782941680205178e-09
      children [
        DEF vWall Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.3
          }
          geometry Box {
            size 0.015 0.065 0.8
          }
        }
      ]
      name "solid(3)"
      boundingObject USE vWall
      physics Physics {
        density 100
      }
      linearVelocity -7.52090552224987e-17 -3.69919929673794e-08 4.268010667150119e-18
      angularVelocity 2.6513024623505304e-16 -4.833467165030712e-18 2.159033281127216e-15
    }
    DEF Pole MazeBlock {
      translation -0.9049999999999255 -1.1475923940712507e-06 -0.577499999999925
      rotation 0.6909121022861573 0.48854282979713215 -0.5328849504051816 -1.922540136207471e-16
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5774999999999252 -1.1475923940712507e-06 -0.2474999999999249
      rotation 0.35728618353353414 0.9274743934638353 -0.11017183181255014 -4.920792817597144e-16
      name "maze block(20)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4124999999999249 -1.1475923940781896e-06 -0.24749999999992506
      rotation 0.21836791281928825 0.955338017920536 -0.19910983442966693 1.6239256086107833e-16
      name "maze block(21)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.24749999999992506 -1.1475923940712507e-06 -0.24749999999992497
      rotation -0.04230369996025734 0.9164101116521816 0.39799862340378617 -3.86216984154952e-16
      name "maze block(22)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.08249999999992501 -1.1475923940712507e-06 -0.24749999999992497
      rotation 0.3182834691012193 0.7615969565553002 -0.564504835287172 3.0760832107020907e-16
      name "maze block(23)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.24750000000007505 -1.1475923940712507e-06 -0.24749999999992497
      rotation -0.9562159117500671 0.03313321804541738 -0.2907805357618399 -2.8467140470634785e-16
      name "maze block(25)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.41250000000007503 -1.1475923940781896e-06 -0.24749999999992506
      rotation 0.21836791281928825 0.955338017920536 -0.19910983442966693 1.6239256086107833e-16
      name "maze block(26)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5774999999999253 -1.1475923940781896e-06 -0.08249999999992502
      rotation -0.7860497050348275 0.5597340584072021 0.262342610099205 -3.133818343598573e-16
      name "maze block(30)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4124999999999248 -1.1475923940712507e-06 -0.08249999999992502
      rotation 0.4931796927611003 0.18129670293676103 0.850826243220275 5.355636032472768e-16
      name "maze block(31)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.247499999999925 -1.1475923940712507e-06 -0.082499999999925
      rotation -0.7094380999170539 0.4693276915980628 0.5257652520710302 -6.788336058274449e-16
      name "maze block(32)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.08249999999992497 -1.1475923940781896e-06 -0.08249999999992493
      rotation -0.9405003590894928 0.09155170078807633 -0.3272267724917164 2.302531602145053e-16
      name "maze block(33)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.08250000000007504 -1.1475923940781896e-06 -0.08249999999992493
      rotation -0.9405003590894928 0.09155170078807633 -0.3272267724917164 2.302531602145053e-16
      name "maze block(34)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.247500000000075 -1.1475923940781896e-06 -0.08249999999992501
      rotation -0.476668964555859 0.2917375127774877 0.8292622756810119 3.719947055443664e-16
      name "maze block(35)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.41250000000007514 -1.1475923940712507e-06 -0.08249999999992502
      rotation 0.4931796927611003 0.18129670293676103 0.850826243220275 5.355636032472768e-16
      name "maze block(36)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.5775000000000747 -1.1475923940781896e-06 -0.08249999999992502
      rotation -0.7860497050348275 0.5597340584072021 0.262342610099205 -3.133818343598573e-16
      name "maze block(37)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5774999999999253 -1.1475923940781896e-06 0.08250000000007499
      rotation -0.7860497050348275 0.5597340584072021 0.262342610099205 -3.133818343598573e-16
      name "maze block(40)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4124999999999248 -1.1475923940712507e-06 0.08250000000007499
      rotation 0.4931796927611003 0.18129670293676103 0.850826243220275 5.355636032472768e-16
      name "maze block(41)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.247499999999925 -1.1475923940712507e-06 0.08250000000007501
      rotation -0.7094380999170539 0.4693276915980628 0.5257652520710302 -6.788336058274449e-16
      name "maze block(42)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.08249999999992497 -1.1475923940781896e-06 0.08250000000007508
      rotation -0.9405003590894928 0.09155170078807633 -0.3272267724917164 2.302531602145053e-16
      name "maze block(43)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.08250000000007504 -1.1475923940781896e-06 0.08250000000007508
      rotation -0.9405003590894928 0.09155170078807633 -0.3272267724917164 2.302531602145053e-16
      name "maze block(44)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.247500000000075 -1.1475923940781896e-06 0.082500000000075
      rotation -0.476668964555859 0.2917375127774877 0.8292622756810119 3.719947055443664e-16
      name "maze block(45)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.41250000000007514 -1.1475923940712507e-06 0.08250000000007499
      rotation 0.4931796927611003 0.18129670293676103 0.850826243220275 5.355636032472768e-16
      name "maze block(46)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.5774999999999254 -1.1475923940781896e-06 0.247500000000075
      rotation 0.28416387282681854 0.9548780239441482 -0.08636464999397896 4.843762207048105e-16
      name "maze block(50)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.4124999999999248 -1.1475923940712507e-06 0.2475000000000752
      rotation 0.30120760732040824 0.9364320064882754 0.17991407537115525 -1.7926791752359129e-16
      name "maze block(51)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.24749999999992506 -1.1475923940712507e-06 0.24750000000007502
      rotation -0.7076735389121828 0.06988160781289633 -0.7030751902983008 5.464619990415543e-16
      name "maze block(52)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation -0.08249999999992495 -1.1475923940712507e-06 0.24750000000007508
      rotation -0.15615086080367208 0.4353840025934817 -0.8865989391804795 1.5258999801066381e-16
      name "maze block(53)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.08250000000007505 -1.1475923940712507e-06 0.24750000000007508
      rotation -0.15615086080367208 0.4353840025934817 -0.8865989391804795 1.5258999801066381e-16
      name "maze block(54)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.24750000000007502 -1.1475923940712507e-06 0.24750000000007508
      rotation 0.2578466093803228 0.8496783641632333 -0.45995848019572727 -4.579917449464987e-16
      name "maze block(55)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.41250000000007514 -1.1475923940712507e-06 0.2475000000000752
      rotation 0.30120760732040824 0.9364320064882754 0.17991407537115525 -1.7926791752359129e-16
      name "maze block(56)"
      length 0.015
    }
    DEF Pole MazeBlock {
      translation 0.5775000000000746 -1.1475923940781896e-06 0.247500000000075
      rotation 0.28416387282681854 0.9548780239441482 -0.08636464999397896 4.843762207048105e-16
      name "maze block(57)"
      length 0.015
    }
    DEF hBlock MazeBlock {
      translation -0.6599999999996645 -1.0710862581475578e-05 -0.5774999999905023
      rotation -2.3126911944061455e-08 0.9999999999999998 -1.4156219324064928e-09 4.7976059656399905e-09
      name "maze block(10)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.3300000000003351 -1.0710862581475578e-05 0.08250000000949821
      rotation 5.4550449360723343e-08 0.9999999999999984 1.826256172924782e-08 4.7976060304571735e-09
      name "maze block(62)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.16500000000033535 -1.0710862581468639e-05 -0.0824999999905018
      rotation -1.6653768453213595e-08 0.9999999999999999 1.1532985412262835e-09 4.797606010213145e-09
      name "maze block(61)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.16500000000033535 -1.0710862581468639e-05 0.08250000000949821
      rotation -1.6653768453213595e-08 0.9999999999999999 1.1532985412262835e-09 4.797606010213145e-09
      name "maze block(60)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.330000000000335 -1.0710862581482516e-05 -0.24749999999050162
      rotation 4.2113355080851715e-08 0.9999999999999992 6.302303126079038e-09 4.797605830419516e-09
      name "maze block(59)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.495000000000335 -1.0710862581475578e-05 -0.08249999999050169
      rotation -4.485579667791032e-08 0.9999999999999989 -7.336114641176071e-09 4.797605990779084e-09
      name "maze block(58)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.4944848629199056 -1.0710862681888311e-05 0.23596830223331508
      rotation -3.6922002343923345e-11 1 4.9985150003008025e-12 -0.030828014712750273
      name "maze block(39)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 0.1650000000003353 -1.0710862581475578e-05 0.2475000000094983
      rotation 8.201480211720937e-09 1 -3.942713659855345e-09 4.797605989464462e-09
      name "maze block(29)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.9049999999905022 -1.0710862581475578e-05 -0.33000000000033525
      rotation -9.431699187358755e-17 1 9.217732080217113e-18 1.5707996986971802
      name "maze block(78)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.41249999999050163 -1.0710862581475578e-05 -3.3545731530091704e-13
      rotation -1.0122598820703214e-16 1 1.8843642990779662e-16 1.5707996986971804
      name "maze block(65)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.24750000000949823 -1.0710862581482516e-05 0.16499999999966464
      rotation -1.7806369432237716e-16 1 6.812506523923549e-17 1.5707996986971804
      name "maze block(64)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.2474999999905017 -1.0710862581468639e-05 -3.354552334075228e-13
      rotation -6.023376365437675e-17 1 -1.4285830015485676e-19 1.5707996986971804
      name "maze block(63)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.24750000000949837 -1.0710862581475578e-05 -0.16500000000033538
      rotation -6.135021558290262e-17 1 9.1232978869638e-17 1.5707996986971806
      name "maze block(49)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.4125000000094985 -1.0710862581475578e-05 -0.1650000000003355
      rotation -1.2142013008475964e-16 1 1.1815999831954388e-16 1.5707996986971802
      name "maze block(11)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation 0.4125000000094982 -1.0710862581468639e-05 0.3299999999996647
      rotation -7.961504127596757e-18 1 1.2774679643406123e-16 1.5707996986971804
      name "maze block(48)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.494999999999665 -1.0710862581475578e-05 -0.24749999999050176
      rotation -5.4543771890694457e-08 0.9999999999999983 -1.416931257735117e-09 4.7976059784699655e-09
      name "maze block(4)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.6599999999996644 -1.0710862581468639e-05 -0.08249999999050173
      rotation -7.032842984764349e-08 0.9999999999999976 -3.950241018118426e-09 4.797605899856741e-09
      name "maze block(6)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 3.354241338601509e-13 -1.0710862581482516e-05 -0.08249999999050175
      rotation 2.9594272754192135e-08 0.9999999999999996 -5.6074632851912425e-09 4.797605859861421e-09
      name "maze block(8)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation 3.354241338601509e-13 -1.0710862581482516e-05 0.08250000000949825
      rotation 2.9594272754192135e-08 0.9999999999999996 -5.6074632851912425e-09 4.797605859861421e-09
      name "maze block(9)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.32999999999966495 -1.0710862581475578e-05 0.08250000000949821
      rotation 5.4550449360723343e-08 0.9999999999999984 1.826256172924782e-08 4.7976060304571735e-09
      name "maze block(12)"
      length 0.14
    }
    DEF hBlock MazeBlock {
      translation -0.32999999999966495 -1.0710862581475578e-05 0.24750000000949837
      rotation 5.379939974135372e-08 0.9999999999999984 2.847048068571022e-09 4.797605914656663e-09
      name "maze block(13)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.08249999999050181 -1.0710862581482516e-05 0.32999999999966484
      rotation -4.114801072546658e-17 1 8.714232453695897e-17 1.5707996986971802
      name "maze block(18)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.08249999999050181 -1.0710862581482516e-05 -0.3300000000003352
      rotation -4.114801072546658e-17 1 8.714232453695897e-17 1.5707996986971802
      name "maze block(2)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.08249999999050164 -1.0710862581475578e-05 -3.3547384911157797e-13
      rotation 1.0376669333562481e-16 1 -1.2310554060353771e-16 1.5707996986971806
      name "maze block(17)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.24749999999050162 -1.0710862581475578e-05 -0.1650000000003354
      rotation -2.74727439836593e-16 1 2.390709033056523e-16 1.5707996986971806
      name "maze block(15)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.5774999999905022 -1.0710862581468639e-05 0.3299999999996647
      rotation 1.7142169175580472e-16 1 -1.0897850641664438e-16 1.5707996986971804
      name "maze block(1)"
      length 0.14
    }
    DEF vBlock MazeBlock {
      translation -0.5774999999905024 -1.0710862581475578e-05 0.16499999999966455
      rotation 1.2964307830317361e-16 1 -1.7756767112696306e-16 1.5707996986971804
      name "maze block(3)"
      length 0.14
    }
  ]
}
